### DISCUSSION TITLE
Possible bug: Trying to use Traefik which utilizes lego, with ZeroSSL and Cloudflare

### DISCUSSION THREAD

**Discussion by samip5** [go-acme/lego#1475] [Q&A]
Hi there,

I'm trying to use CLOUDFLARE_API_KEY and CLOUDFLARE_EMAIL env variables, to get this to work, but seems like Cloudflare is giving errors?

I'm not sure if this is related to Cloudflare's API itself because that samip.dev with the ending dot is incorrect.

`acme: cleaning up failed: cloudflare: failed to find zone samip.dev.: ListZonesContext command failed: HTTP status 400: Invalid request headers (6003)`

**Comment by prabirshrestha** [comment]
I'm getting similar issues. I made sure I have appropriate access as mentioned in the docs with `Zone / Zone / Read` and `Zone / DNS / Edit`. I have also tried with global api key and get the same error. 

I see the same trailing dot on my domain.
